En este cuaderno se intentará ver los accidentes ocurridos cerca de colegios y escuelas

In [66]:
import chardet
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from my_map_creator import MapCreator
import my_map_creator

df_accidentes = pd.read_csv('./Datasets/Data_Combinada.csv')
df_schools = pd.read_csv('./Datasets/212790-0-centros-educacion.csv', sep=';', encoding='ISO-8859-1')

/tmp/ipykernel_62465/2967255395.py:8: DtypeWarning: Columns (15,16) have mixed types. Specify dtype option on import or set low_memory=False.
  df_accidentes = pd.read_csv('./Datasets/Data_Combinada.csv')


In [67]:
with open('./Datasets/212790-0-centros-educacion.csv', 'rb') as f:
    result = chardet.detect(f.readline())
    print(result)

{'encoding': 'ascii', 'confidence': 1.0, 'language': ''}


# Análisis Exploratorio Inicial

In [68]:
df_schools.head()

PK                                             NOMBRE  \
0    167390                             Academia Darío Estudio   
1      5373                      Altair, Colegio Internacional   
2  10127337  Aula Municipal de Formación de Personas Adulta...   
3   5422896  Aula Municipal de Formación de Personas Adulta...   
4      6599  Aula Municipal de Formación de Personas Adulta...   

  DESCRIPCION-ENTIDAD                                            HORARIO  \
0                 NaN                                                NaN   
1                 NaN                                                NaN   
2                 NaN  De lunes a jueves:    Mañanas de 08:30 a 13:00...   
3                 NaN  De lunes a jueves:  mañanas de 10,15 a 12,15 h...   
4                 NaN  De lunes a jueves:   mañanas de 9:00 a 13:30 h...   

                              EQUIPAMIENTO  \
0                                      NaN   
1  Comedor - Transporte - Horario ampliado   
2                                      NaN   
3                                      NaN   
4                                      NaN   

                                          TRANSPORTE  \
0  METRO: Santo Domingo, Plaza España BUS: 1 , 2 ...   
1  METRO: Plaza de Castilla BUS: 5 , 27 , 47 , 49...   
2  Metro: Embajadores. Lavapiés, Acacias    Bus: ...   
3       Bus: 3, 7, 21, 37, 40 Metro: Alonso Martínez   
4  Bus: 15 , 28 , 110 , 113 , 210 Metro: La Elipa...   

                                         DESCRIPCION  ACCESIBILIDAD  \
0  ENSEÑANZA: BACHILLERATO (Humanidades  y C. Soc...            0.0   
1  ENSEÑANZA: Educación infantil - Primaria - ESO...            0.0   
2  Programas Formativos:   Español para extranjeros.            1.0   
3  Programas Formativos:   LectoescrituraMatemáti...            0.0   
4  Programas Formativos:    Lectoescritura. Matem...            1.0   

                                         CONTENT-URL       NOMBRE-VIA  ...  \
0  http://www.madrid.es/sites/v/index.jsp?vgnextc...         GRAN VIA  ...   
1  http://www.madrid.es/sites/v/index.jsp?vgnextc...      JOAQUIN BAU  ...   
2  http://www.madrid.es/sites/v/index.jsp?vgnextc...           CASINO  ...   
3  http://www.madrid.es/sites/v/index.jsp?vgnextc...          ALMAGRO  ...   
4  http://www.madrid.es/sites/v/index.jsp?vgnextc...  SANTA FELICIDAD  ...   

  COD-DISTRITO       DISTRITO COORDENADA-X COORDENADA-Y    LATITUD  LONGITUD  \
0          1.0         CENTRO     439895.0    4475101.0  40.422491 -3.709818   
1          5.0      CHAMARTIN     441854.0    4479461.0  40.463788 -3.685848   
2          1.0         CENTRO     440152.0    4473128.0  40.406617 -3.705326   
3          7.0       CHAMBERI     441045.0    4475554.0  40.428526 -3.695039   
4         15.0  CIUDAD LINEAL     444837.0    4475071.0  40.424435 -3.650284   

                    TELEFONO          FAX                         EMAIL  \
0  915 411 515 - 915 474 084  915 474 084          info@darioestudio.es   
1  913 451 168 - 913 596 368  913 507 016  secretaria@colegioaltair.com   
2               91 539 63 10          NaN                           NaN   
3  917 001 548 - 917 001 552          NaN                           NaN   
4                914 055 312          NaN                           NaN   

                                                TIPO  
0  /contenido/entidadesYorganismos/CentrosEducaci...  
1  /contenido/entidadesYorganismos/CentrosEducaci...  
2  /contenido/entidadesYorganismos/CentrosEducaci...  
3  /contenido/entidadesYorganismos/CentrosEducaci...  
4  /contenido/entidadesYorganismos/CentrosEducaci...  

[5 rows x 32 columns]

In [69]:
df1_info = df_schools.info()

df1_description = df_schools.describe(include='all')

df1_nulls = df_schools.isnull().sum()

df1_duplicated = df_schools.duplicated().sum()

df1_info, df1_description, df1_nulls, df1_duplicated

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1503 entries, 0 to 1502
Data columns (total 32 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   PK                   1503 non-null   int64  
 1   NOMBRE               1503 non-null   object 
 2   DESCRIPCION-ENTIDAD  90 non-null     object 
 3   HORARIO              25 non-null     object 
 4   EQUIPAMIENTO         788 non-null    object 
 5   TRANSPORTE           1489 non-null   object 
 6   DESCRIPCION          1460 non-null   object 
 7   ACCESIBILIDAD        1440 non-null   float64
 8   CONTENT-URL          1503 non-null   object 
 9   NOMBRE-VIA           1503 non-null   object 
 10  CLASE-VIAL           1502 non-null   object 
 11  TIPO-NUM             1501 non-null   object 
 12  NUM                  1490 non-null   object 
 13  PLANTA               4 non-null      object 
 14  PUERTA               0 non-null      float64
 15  ESCALERAS            0 non-null      f

(None,
                   PK                  NOMBRE  \
 count   1.503000e+03                    1503   
 unique           NaN                    1503   
 top              NaN  Academia Darío Estudio   
 freq             NaN                       1   
 mean    2.350380e+06                     NaN   
 std     3.733567e+06                     NaN   
 min     1.085000e+03                     NaN   
 25%     5.543500e+03                     NaN   
 50%     6.227000e+03                     NaN   
 75%     5.154940e+06                     NaN   
 max     1.208375e+07                     NaN   
 
                                       DESCRIPCION-ENTIDAD             HORARIO  \
 count                                                  90                  25   
 unique                                                 12                  18   
 top     Centro Privado de Formación Profesional Especí...  Jornada continuada   
 freq                                                   29                 

### Estructura del Dataset:
El dataset tiene 1503 filas y 32 columnas.
Las columnas cubren una amplia gama de información, desde detalles del centro educativo (nombre, descripción, horario, etc.) hasta información de contacto y ubicación geográfica.

### Calidad de los Datos:

Hay una cantidad significativa de valores faltantes en varias columnas, especialmente en DESCRIPCION-ENTIDAD, HORARIO, PLANTA, PUERTA, ESCALERAS, y ORIENTACION. Esto indica que ciertas columnas pueden no ser útiles para un análisis más profundo a menos que se realice una limpieza de datos.
La columna ACCESIBILIDAD parece ser numérica con valores 0 o 1, indicando si el centro es accesible o no.
No se detectaron filas duplicadas, lo que es positivo para la integridad del dataset.
Las columnas LATITUD y LONGITUD proporcionan ubicaciones geográficas, que utilizaremos para nuestro posterior análisis.

### Descripción de las Columnas:

Cada centro educativo tiene un NOMBRE único.
TIPO tiene 13 categorías únicas, indicando posiblemente tipos diferentes de centros educativos.
EMAIL tiene 803 valores únicos, lo que indica que no todos los centros tienen información de contacto por correo electrónico disponible.

In [70]:
df_map = df_schools.filter(items=['NOMBRE', 'COORDENADA-X', 'COORDENADA-Y', 'LONGITUD', 'LATITUD'])
df_map.dropna()
df_map.head()

NOMBRE  COORDENADA-X  \
0                             Academia Darío Estudio      439895.0   
1                      Altair, Colegio Internacional      441854.0   
2  Aula Municipal de Formación de Personas Adulta...      440152.0   
3  Aula Municipal de Formación de Personas Adulta...      441045.0   
4  Aula Municipal de Formación de Personas Adulta...      444837.0   

   COORDENADA-Y  LONGITUD    LATITUD  
0     4475101.0 -3.709818  40.422491  
1     4479461.0 -3.685848  40.463788  
2     4473128.0 -3.705326  40.406617  
3     4475554.0 -3.695039  40.428526  
4     4475071.0 -3.650284  40.424435

Procedemos a pintar los colegios sobre el mapa para ver si está bien clasificados

In [71]:
map_schools = MapCreator(df_map, lat_col_name='LATITUD', long_col_name='LONGITUD',
                         columns_for_label=['NOMBRE', 'LATITUD', 'LONGITUD'],
                         is_cords_utm=False)
map_schools.create_map(map_name='colegios')

# Limpieza de datos
Vamos a utilizar la información de la calle para agrupar los accidentes, por lo que es necesario limpiar esas columnas

In [72]:
df_schools

PK                                             NOMBRE  \
0       167390                             Academia Darío Estudio   
1         5373                      Altair, Colegio Internacional   
2     10127337  Aula Municipal de Formación de Personas Adulta...   
3      5422896  Aula Municipal de Formación de Personas Adulta...   
4         6599  Aula Municipal de Formación de Personas Adulta...   
...        ...                                                ...   
1498      6258          Organización Profesional Española (OPESA)   
1499      6070            Real Colegio Santa Isabel - La Asunción   
1500   6460544           San Román Escuela de Estudios Superiores   
1501   7531916  Sanromán Centro Internacional de Estudios Prof...   
1502     30856                      Trainek formación profesional   

                                    DESCRIPCION-ENTIDAD  \
0                                                   NaN   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
1498  Centro privado de formación profesional especí...   
1499                                                NaN   
1500  Centro privado de formación profesional especí...   
1501  Centro privado de formación profesional especí...   
1502  Centro Privado de Formación Profesional Especí...   

                                                HORARIO  \
0                                                   NaN   
1                                                   NaN   
2     De lunes a jueves:    Mañanas de 08:30 a 13:00...   
3     De lunes a jueves:  mañanas de 10,15 a 12,15 h...   
4     De lunes a jueves:   mañanas de 9:00 a 13:30 h...   
...                                                 ...   
1498                                                NaN   
1499                                                NaN   
1500                                                NaN   
1501                                                NaN   
1502                                                NaN   

                                 EQUIPAMIENTO  \
0                                         NaN   
1     Comedor - Transporte - Horario ampliado   
2                                         NaN   
3                                         NaN   
4                                         NaN   
...                                       ...   
1498                                      NaN   
1499               Horario ampliado - Comedor   
1500                                      NaN   
1501                                      NaN   
1502                                      NaN   

                                             TRANSPORTE  \
0     METRO: Santo Domingo, Plaza España BUS: 1 , 2 ...   
1     METRO: Plaza de Castilla BUS: 5 , 27 , 47 , 49...   
2     Metro: Embajadores. Lavapiés, Acacias    Bus: ...   
3          Bus: 3, 7, 21, 37, 40 Metro: Alonso Martínez   
4     Bus: 15 , 28 , 110 , 113 , 210 Metro: La Elipa...   
...                                                 ...   
1498  Metro: Manuel Becerra (líneas 2 y 6).  Bus: 14...   
1499  Metro: Atocha , Antón Martín  Bus: 6 , 19 , 26...   
1500  Metro: Lista (línea 4), Manuel Becerra (líneas...   
1501  Metro: Diego de León (línea 4, 5 y 6).  Bus: 4...   
1502                                                NaN   

                                            DESCRIPCION  ACCESIBILIDAD  \
0     ENSEÑANZA: BACHILLERATO (Humanidades  y C. Soc...            0.0   
1     ENSEÑANZA: Educación infantil - Primaria - ESO...            0.0   
2     Programas Formativos:   Español para extranjeros.            1.0   
3     Programas Formativos:   LectoescrituraMatemáti...            0.0   
4     Programas Formativos:    Lectoescritura. Matem...            1.0   
...                    

Tiramos las columnas con un exceso de nulos. Es decir, más nulos que valores, y que además, no conciernen a la investigación, ya que solo necesitamos información sobre la dirección de la escuela.

Descripción entidad tiene una calidad demasiado mala, como se ve abajo.
El equipamiento de la escuela no nos importa.
El horario tiene una calidad demasiada mala
El transporte no nos importa.
La descripción puede servir para saber la edad de los implicados.
La accesibilidad no importa al análisis a realizar.
El url no importa.
Nombre de vía y clase vial a quedarse.
Tipo num es de tipo V/KM o Sin número, en principio no importa ya que solo vamos a quedarnos con la calle.
Planta tiene 4 entradas de 1503, a eliminar.
Escaleras no tiene ninguna entrada.
Orientación tiene el 96.4% de sus valores nulos, lo exploraré más abajo.
La localidad y la provincia es siempre Madrid, la columna se tira.
El código postal será útil para agrupar por distritos.
El código barrio no aparece en la documentación, pero teniendo codigo distrito será innecesario.
Distrito y coordenadas a quedarse.
Orientación a eliminar, según el anexo es solo para encontrar más fácilmente el edificio.
Los datos de contacto no son necesarios.
El tipo de escuela es interesante, puede servir para clasificar a adultos niños y adolescentes.


In [73]:
df_schools['DESCRIPCION-ENTIDAD'].unique()

array([nan, 'Centro Privado de Formación Profesional Específica.',
       'Fundación Querer',
       'Centro Privado de Formación Profesional Específica',
       'Centro privado de formación profesional específica',
       'Centro privado de formación profesional específica.',
       'Centro privado de formacion profesional específica',
       'Centro privado de formacion profesional específica.',
       'Centro Privado en Formación Profesional específica.',
       'Centro Privado en Formación Profesional específica',
       'Centro privado de educación secundaria',
       'ENSEÑANZA: Educacion Infantil - Primaria - Enseñanza biling&amp;uuml;e',
       'Educación de 0 a 3 años'], dtype=object)

Calidad malísima, no merece la pena recuperar, igual que horario, que tiene solo 25 entradas de 1503

In [74]:
df_schools.drop(
    columns=['DESCRIPCION-ENTIDAD', 'HORARIO', 'TRANSPORTE', 'PK', 'DESCRIPCION-ENTIDAD', 'CONTENT-URL', 'EMAIL',
             'EQUIPAMIENTO', 'ESCALERAS', 'FAX', 'ACCESIBILIDAD', 'PUERTA', 'PLANTA', 'LOCALIDAD', 'PROVINCIA',
             'COD-BARRIO', 'TELEFONO', 'ORIENTACION'], inplace=True)

In [75]:
df_schools

NOMBRE  \
0                                Academia Darío Estudio   
1                         Altair, Colegio Internacional   
2     Aula Municipal de Formación de Personas Adulta...   
3     Aula Municipal de Formación de Personas Adulta...   
4     Aula Municipal de Formación de Personas Adulta...   
...                                                 ...   
1498          Organización Profesional Española (OPESA)   
1499            Real Colegio Santa Isabel - La Asunción   
1500           San Román Escuela de Estudios Superiores   
1501  Sanromán Centro Internacional de Estudios Prof...   
1502                      Trainek formación profesional   

                                            DESCRIPCION          NOMBRE-VIA  \
0     ENSEÑANZA: BACHILLERATO (Humanidades  y C. Soc...            GRAN VIA   
1     ENSEÑANZA: Educación infantil - Primaria - ESO...         JOAQUIN BAU   
2     Programas Formativos:   Español para extranjeros.              CASINO   
3     Programas Formativos:   LectoescrituraMatemáti...             ALMAGRO   
4     Programas Formativos:    Lectoescritura. Matem...     SANTA FELICIDAD   
...                                                 ...                 ...   
1498  Formación profesional de grado superior  Sanid...               AYALA   
1499  ENSEÑANZA: Educación Infantil (segundo ciclo) ...        SANTA ISABEL   
1500  Formación profesional de grado  superior   Ser...             MONTESA   
1501  Formación profesional grado medio  Servicios s...      FERRER DEL RIO   
1502  Formación profesional grado medio  Electricida...  MARIO ROSO DE LUNA   

     CLASE-VIAL TIPO-NUM  NUM  CODIGO-POSTAL        BARRIO  COD-DISTRITO  \
0         CALLE        V   63          28013       PALACIO           1.0   
1         CALLE      NUM    4          28036  NUEVA ESPAÑA           5.0   
2         CALLE        V    5          28005   EMBAJADORES           1.0   
3         CALLE        V    5          28010       ALMAGRO           7.0   
4         CALLE      NUM   39          28017        VENTAS          15.0   
...         ...      ...  ...            ...           ...           ...   
1498      CALLE        V  111          28009          GOYA           4.0   
1499      CALLE        V   46          28012   EMBAJADORES           1.0   
1500      CALLE        V   39          28006         LISTA           4.0   
1501      CALLE        V   10          28028    GUINDALERA           4.0   
1502      CALLE        V   29          28022         REJAS          20.0   

                 DISTRITO  COORDENADA-X  COORDENADA-Y    LATITUD  LONGITUD  \
0                  CENTRO      439895.0     4475101.0  40.422491 -3.709818   
1               CHAMARTIN      441854.0     4479461.0  40.463788 -3.685848   
2                  CENTRO      440152.0     4473128.0  40.406617 -3.705326   
3                CHAMBERI      441045.0     4475554.0  40.428526 -3.695039   
4           CIUDAD LINEAL      444837.0     4475071.0  40.424435 -3.650284   
...                   ...           ...           ...        ...       ...   
1498            SALAMANCA      443197.0     4475378.0  40.427106 -3.669642   
1499               CENTRO      441006.0     4473664.0  40.409625 -3.696577   
1500            SALAMANCA      442979.0     4475750.0  40.430437 -3.672239   
1501            SALAMANCA      442962.0     4476143.0  40.433978 -3.672479   
1502  SAN BLAS-CANILLEJAS      452533.0     4477166.0  40.443793 -3.559715   

                                                   TIPO  
0     /contenido/entidadesYorganismos/CentrosEducaci...  
1     /contenido/entidadesYorganismos/CentrosEducaci...  
2     /contenido/entidadesYorganismos/CentrosEducaci...  
3     /contenido/entidadesYorganismos/CentrosEducaci...  
4     /contenido/entidadesYorganismos/CentrosEducaci...  
...                                                 ...  
1498  /contenido/entidadesYorganismos/CentrosEducaci...  
1499  /contenido/entidadesYorganismos/MonumentosEdif...  
1500  /contenido/en

In [76]:
df_schools['DESCRIPCION'].unique()

array(['ENSEÑANZA: BACHILLERATO (Humanidades  y C. Sociales - Ciencias y Tecnología)',
       'ENSEÑANZA: Educación infantil - Primaria - ESO - BACHILLERATO (Humanidades y Ciencias Sociales - Ciencias)',
       'Programas Formativos:   Español para extranjeros.',
       'Programas Formativos:   LectoescrituraMatemáticas para la vidaEspañol para inmigrantes',
       'Programas Formativos:    Lectoescritura. Matemáticas para la vida. Español para extranjeros. Alfabetización digital.',
       'Programas Formativos:   Lectoescritura.Matemáticas para la vida.',
       'ENSEÑANZA: Educación infantil (Primer Ciclo)',
       'ENSEÑANZA: Educación infantil (Primer ciclo)',
       'ENSEÑANZA: Educación Infantil (Primer ciclo)',
       'Enseñanzas  Educación especial:  Educación Básica Obligatoria',
       'Enseñanzas   Enseñanzas Iniciales Básicas para personas adultas Educación Secundaria Obligatoria para personas adultas',
       'Formación profesional de grado medio  Administración y gestión:

Vamos a ver los 43 faltantes de descripción

In [77]:
df_schools[df_schools['DESCRIPCION'].isna()]

NOMBRE DESCRIPCION  \
19    Centro de Educación de Personas Adultas (CEPA)...         NaN   
20    Centro de Educación de Personas Adultas (CEPA)...         NaN   
21    Centro de Educación de Personas Adultas (CEPA)...         NaN   
22    Centro de Educación de Personas Adultas (CEPA)...         NaN   
23    Centro de Educación de Personas Adultas (CEPA)...         NaN   
24    Centro de Educación de Personas Adultas (CEPA)...         NaN   
25    Centro de Educación de Personas Adultas (CEPA)...         NaN   
27    Centro de Educación de Personas Adultas (CEPA)...         NaN   
28    Centro de Educación de Personas Adultas (CEPA)...         NaN   
29    Centro de Educación de Personas Adultas (CEPA)...         NaN   
30    Centro de Educación de Personas Adultas (CEPA)...         NaN   
32    Centro de Educación de Personas Adultas (CEPA)...         NaN   
33    Centro de Educación de Personas Adultas (CEPA)...         NaN   
34    Centro de Educación de Personas Adultas (CEPA)...         NaN   
35    Centro de Educación de Personas Adultas (CEPA)...         NaN   
36    Centro de Educación de Personas Adultas (CEPA)...         NaN   
37    Centro de Educación de Personas Adultas (CEPA)...         NaN   
39    Centro de Educación de Personas Adultas (CEPA)...         NaN   
40    Centro de Educación de Personas Adultas (CEPA)...         NaN   
41    Centro de Educación de Personas Adultas (CEPA)...         NaN   
42    Centro de Educación de Personas Adultas (CEPA)...         NaN   
43    Centro de Educación de Personas Adultas (CEPA)...         NaN   
44    Centro de Educación de Personas Adultas (CEPA)...         NaN   
45    Centro de Educación de Personas Adultas (CEPA)...         NaN   
89                           Centro de enseñanza CODESA         NaN   
90                        Centro de enseñanza Cruz Roja         NaN   
91    Centro de enseñanza Cruz Roja Escuela Profesional         NaN   
146   Centro integrado de enseñanzas regladas a dist...         NaN   
154   Centro Privado de Educación de Personas Adulta...         NaN   
156   Centro Privado de Educación de Personas Adulta...         NaN   
158   Centro Privado de Educación de Personas Adulta...         NaN   
903               Escuela Infantil Contumami la Vaguada         NaN   
947   Escuela Infantil El Bosque Encantado. Calle An...         NaN   
988                        Escuela Infantil Fantasía II         NaN   
1057  Escuela Infantil Lapizitos. Calle Condesa de Teba         NaN   
1102                     Escuela Infantil Mamá Queca II         NaN   
1225                             Escuela Infantil Nubbe         NaN   
1227     Escuela Infantil Nubes. Calle Gabriel Usera 39         NaN   
1250                          Escuela Infantil Pequemun         NaN   
1256                   Escuela Infantil Pequeños Amigos         NaN   
1317   Escuela Infantil Smiles Bilingual Nursery School         NaN   
1338                          Escuela Infantil Trapitos         NaN   
1369                       ESERP Madrid Business School         NaN   

                               NOMBRE-VIA CLASE-VIAL TIPO-NUM  NUM  \
19             CORREGIDOR ALONSO DE TOBAR      CALLE        V    7   
20                       OCA&amp;Ntilde;A      CALLE        V   35   
21                     BATALLA DEL SALADO      CALLE        V   29   
22                                  MUSAS      CALLE        V   11   
23                        ANSELMO LORENZO      CALLE        V   12   
24               HERMANOS GARCIA NOBLEJAS      CALLE        V   70   
25                         FERNANDO ORTIZ      CALLE        V    2   
27                               ABERTURA      CALLE        V    2   
28                                 SERENA      CALLE        V  394   
29                                BRAILLE      CALLE        V   10   
30                           MAR AMARILLO      CALLE        V   21   
32                               CHANTADA      CALLE        V   34   
33    NUESTRA 

Vamos a indagar a ver si alguno tiene puesta la etiqueta

In [78]:
patron = r"Personas Adultas"

df_coincidencias = df_schools[df_schools['NOMBRE'].str.contains(patron)].copy()
df_coincidencias['DESCRIPCION'].unique()

array(['Programas Formativos:   Español para extranjeros.',
       'Programas Formativos:   LectoescrituraMatemáticas para la vidaEspañol para inmigrantes',
       'Programas Formativos:    Lectoescritura. Matemáticas para la vida. Español para extranjeros. Alfabetización digital.',
       'Programas Formativos:   Lectoescritura.Matemáticas para la vida.',
       nan,
       'Enseñanzas:   Preparación acceso a la universidad para mayores de 25 años Preparación para  la prueba de conocimientos constitucionales Talleres ocupacionales cofinanciados Talleres operativos para alumnado con discapacidad Enseñanzas iniciales básicas para personas adultas Educación Secundaria obligatoria para personas adultas Español para extranjeros Cursos a distancia soporte telemático: Aula Mentor',
       'Enseñanzas:   Francés Inglés Informática Informática-ofimática Escritura creativa Preparación para la prueba de conocimientos constitucionales Preparación acceso Ciclos Formativos Grado Superior Enseñanzas

In [79]:
df_coincidencias

NOMBRE  \
2    Aula Municipal de Formación de Personas Adulta...   
3    Aula Municipal de Formación de Personas Adulta...   
4    Aula Municipal de Formación de Personas Adulta...   
5    Aula Municipal de Formación de Personas Adulta...   
6    Aula Municipal de Formación de Personas Adulta...   
7    Aula Municipal de Formación de Personas Adulta...   
19   Centro de Educación de Personas Adultas (CEPA)...   
20   Centro de Educación de Personas Adultas (CEPA)...   
21   Centro de Educación de Personas Adultas (CEPA)...   
22   Centro de Educación de Personas Adultas (CEPA)...   
23   Centro de Educación de Personas Adultas (CEPA)...   
24   Centro de Educación de Personas Adultas (CEPA)...   
25   Centro de Educación de Personas Adultas (CEPA)...   
26   Centro de Educación de Personas Adultas (CEPA)...   
27   Centro de Educación de Personas Adultas (CEPA)...   
28   Centro de Educación de Personas Adultas (CEPA)...   
29   Centro de Educación de Personas Adultas (CEPA)...   
30   Centro de Educación de Personas Adultas (CEPA)...   
31   Centro de Educación de Personas Adultas (CEPA)...   
32   Centro de Educación de Personas Adultas (CEPA)...   
33   Centro de Educación de Personas Adultas (CEPA)...   
34   Centro de Educación de Personas Adultas (CEPA)...   
35   Centro de Educación de Personas Adultas (CEPA)...   
36   Centro de Educación de Personas Adultas (CEPA)...   
37   Centro de Educación de Personas Adultas (CEPA)...   
38   Centro de Educación de Personas Adultas (CEPA)...   
39   Centro de Educación de Personas Adultas (CEPA)...   
40   Centro de Educación de Personas Adultas (CEPA)...   
41   Centro de Educación de Personas Adultas (CEPA)...   
42   Centro de Educación de Personas Adultas (CEPA)...   
43   Centro de Educación de Personas Adultas (CEPA)...   
44   Centro de Educación de Personas Adultas (CEPA)...   
45   Centro de Educación de Personas Adultas (CEPA)...   
154  Centro Privado de Educación de Personas Adulta...   
155  Centro Privado de Educación de Personas Adulta...   
156  Centro Privado de Educación de Personas Adulta...   
157  Centro Privado de Educación de Personas Adulta...   
158  Centro Privado de Educación de Personas Adulta...   
159  Centro Privado de Educación de Personas Adulta...   

                                           DESCRIPCION  \
2    Programas Formativos:   Español para extranjeros.   
3    Programas Formativos:   LectoescrituraMatemáti...   
4    Programas Formativos:    Lectoescritura. Matem...   
5    Programas Formativos:   Lectoescritura.Matemát...   
6    Programas Formativos:   Español para extranjeros.   
7    Programas Formativos:   LectoescrituraMatemáti...   
19                                                 NaN   
20                                                 NaN   
21                                                 NaN   
22                                                 NaN   
23                                                 NaN   
24                                                 NaN   
25                                                 NaN   
26   Enseñanzas:   Preparación acceso a la universi...   
27                                                 NaN   
28                                                 NaN   
29                                                 NaN   
30                                                 NaN   
31   Enseñanzas:   Francés Inglés Informática Infor...   
32                                                 NaN   
33                                                 NaN   
34                                                 NaN   
35                                                 NaN   
36                                                 NaN   
37                                                 NaN   
38   Enseñanzas:   Inglés Ampliación cultural Prepa...   
39                                                 NaN   
40                                                 NaN   
41                                           

Concluimos que no es posible discernirlos con solo su nombre. Pondremos un filler: Centro de educación de adultos

In [80]:
df_coincidencias.fillna('Adultos enseñanzas no especificadas', inplace=True)

/tmp/ipykernel_62465/791856484.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'Adultos enseñanzas no especificadas' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df_coincidencias.fillna('Adultos enseñanzas no especificadas', inplace=True)


In [81]:
df_coincidencias

NOMBRE  \
2    Aula Municipal de Formación de Personas Adulta...   
3    Aula Municipal de Formación de Personas Adulta...   
4    Aula Municipal de Formación de Personas Adulta...   
5    Aula Municipal de Formación de Personas Adulta...   
6    Aula Municipal de Formación de Personas Adulta...   
7    Aula Municipal de Formación de Personas Adulta...   
19   Centro de Educación de Personas Adultas (CEPA)...   
20   Centro de Educación de Personas Adultas (CEPA)...   
21   Centro de Educación de Personas Adultas (CEPA)...   
22   Centro de Educación de Personas Adultas (CEPA)...   
23   Centro de Educación de Personas Adultas (CEPA)...   
24   Centro de Educación de Personas Adultas (CEPA)...   
25   Centro de Educación de Personas Adultas (CEPA)...   
26   Centro de Educación de Personas Adultas (CEPA)...   
27   Centro de Educación de Personas Adultas (CEPA)...   
28   Centro de Educación de Personas Adultas (CEPA)...   
29   Centro de Educación de Personas Adultas (CEPA)...   
30   Centro de Educación de Personas Adultas (CEPA)...   
31   Centro de Educación de Personas Adultas (CEPA)...   
32   Centro de Educación de Personas Adultas (CEPA)...   
33   Centro de Educación de Personas Adultas (CEPA)...   
34   Centro de Educación de Personas Adultas (CEPA)...   
35   Centro de Educación de Personas Adultas (CEPA)...   
36   Centro de Educación de Personas Adultas (CEPA)...   
37   Centro de Educación de Personas Adultas (CEPA)...   
38   Centro de Educación de Personas Adultas (CEPA)...   
39   Centro de Educación de Personas Adultas (CEPA)...   
40   Centro de Educación de Personas Adultas (CEPA)...   
41   Centro de Educación de Personas Adultas (CEPA)...   
42   Centro de Educación de Personas Adultas (CEPA)...   
43   Centro de Educación de Personas Adultas (CEPA)...   
44   Centro de Educación de Personas Adultas (CEPA)...   
45   Centro de Educación de Personas Adultas (CEPA)...   
154  Centro Privado de Educación de Personas Adulta...   
155  Centro Privado de Educación de Personas Adulta...   
156  Centro Privado de Educación de Personas Adulta...   
157  Centro Privado de Educación de Personas Adulta...   
158  Centro Privado de Educación de Personas Adulta...   
159  Centro Privado de Educación de Personas Adulta...   

                                           DESCRIPCION  \
2    Programas Formativos:   Español para extranjeros.   
3    Programas Formativos:   LectoescrituraMatemáti...   
4    Programas Formativos:    Lectoescritura. Matem...   
5    Programas Formativos:   Lectoescritura.Matemát...   
6    Programas Formativos:   Español para extranjeros.   
7    Programas Formativos:   LectoescrituraMatemáti...   
19                 Adultos enseñanzas no especificadas   
20                 Adultos enseñanzas no especificadas   
21                 Adultos enseñanzas no especificadas   
22                 Adultos enseñanzas no especificadas   
23                 Adultos enseñanzas no especificadas   
24                 Adultos enseñanzas no especificadas   
25                 Adultos enseñanzas no especificadas   
26   Enseñanzas:   Preparación acceso a la universi...   
27                 Adultos enseñanzas no especificadas   
28                 Adultos enseñanzas no especificadas   
29                 Adultos enseñanzas no especificadas   
30                 Adultos enseñanzas no especificadas   
31   Enseñanzas:   Francés Inglés Informática Infor...   
32                 Adultos enseñanzas no especificadas   
33                 Adultos enseñanzas no especificadas   
34                 Adultos enseñanzas no especificadas   
35                 Adultos enseñanzas no especificadas   
36                 Adultos enseñanzas no especificadas   
37                 Adultos enseñanzas no especificadas   
38   Enseñanzas:   Inglés Ampliación cultural Prepa...   
39                 Adultos enseñanzas no especificadas   
40                 Adultos enseñanzas no especificadas   
41                 Adultos enseñanzas no espe

In [82]:
df_schools.update(df_coincidencias)

In [83]:
df_schools[df_schools['DESCRIPCION'].isna()]

NOMBRE DESCRIPCION  \
89                           Centro de enseñanza CODESA         NaN   
90                        Centro de enseñanza Cruz Roja         NaN   
91    Centro de enseñanza Cruz Roja Escuela Profesional         NaN   
146   Centro integrado de enseñanzas regladas a dist...         NaN   
903               Escuela Infantil Contumami la Vaguada         NaN   
947   Escuela Infantil El Bosque Encantado. Calle An...         NaN   
988                        Escuela Infantil Fantasía II         NaN   
1057  Escuela Infantil Lapizitos. Calle Condesa de Teba         NaN   
1102                     Escuela Infantil Mamá Queca II         NaN   
1225                             Escuela Infantil Nubbe         NaN   
1227     Escuela Infantil Nubes. Calle Gabriel Usera 39         NaN   
1250                          Escuela Infantil Pequemun         NaN   
1256                   Escuela Infantil Pequeños Amigos         NaN   
1317   Escuela Infantil Smiles Bilingual Nursery School         NaN   
1338                          Escuela Infantil Trapitos         NaN   
1369                       ESERP Madrid Business School         NaN   

                     NOMBRE-VIA CLASE-VIAL TIPO-NUM  NUM  CODIGO-POSTAL  \
89                DIEGO DE LEON      CALLE        V   34          28006   
90              DE RAFAEL VILLA      CALLE        V  NaN          28023   
91               REINA VICTORIA    AVENIDA        V   28          28003   
146                       PRADO      PASEO        V   28          28014   
903                  FONSAGRADA      PLAZA      NUM   10          28029   
947               ANTONIO LOPEZ      CALLE        V  246          28026   
988                JUSTA GARCIA      CALLE        V   12          28026   
1057            CONDESA DE TEBA      CALLE        V   11          28044   
1102           SANCHEZ GUERRERO      CALLE        V    7          28043   
1225                TRES OLIVOS      PLAZA      NUM   16          28034   
1227              GABRIEL USERA      CALLE        V   39          28026   
1250                     OLIVOS      CALLE        V    8          28003   
1256  MONASTERIO DE EL ESCORIAL    AVENIDA      NUM   74          28049   
1317            PUENTE LA REINA      CALLE        V   36          28050   
1338                   LIMONITA      CALLE        V    1          28021   
1369                 COSTA RICA      CALLE        V    9          28016   

                    BARRIO COD-DISTRITO             DISTRITO  COORDENADA-X  \
89              CASTELLANA          4.0            SALAMANCA      442210.0   
90              EL PLANTIO          9.0      MONCLOA-ARAVACA      430541.0   
91    CIUDAD UNIVERSITARIA          9.0      MONCLOA-ARAVACA      439817.0   
146                 CORTES          1.0               CENTRO      441174.0   
903               EL PILAR          8.0  FUENCARRAL-EL PARDO      439960.0   
947            ALMENDRALES         12.0                USERA      440926.0   
988             PRADOLONGO         12.0                USERA      439780.0   
1057            BUENAVISTA         11.0          CARABANCHEL      435616.0   
1102              CANILLAS         16.0            HORTALEZA      444636.0   
1225              VALVERDE          8.0  FUENCARRAL-EL PARDO      441536.0   
1227            PRADOLONGO         12.0                USERA      439858.0   
1250          VALLEHERMOSO          7.0             CHAMBERI      439198.0   
1256             EL GOLOSO          8.0  FUENCARRAL-EL PARDO      441218.0   
1317              VALVERDE          8.0  FUENCARRAL-EL PARDO      442593.0   
1338              BUTARQUE         17.0           VILLAVERDE      442684.0   
1369          NUEVA ESPAÑA          5.0            CHAMARTIN      442773.0   

      COORDENADA-Y    LATITUD  LONGITUD  \
89       4476230.0  40.434707 -3.681346   
90       4480174.0  40.469333 -3.819366   
91       4477640.0  40.447246 -3.709695   
146      4473706.0  40.411896 -3.693334   
903      4481061.0  40.

In [84]:
patron = r"Infantil"

df_coincidencias = df_schools[df_schools['NOMBRE'].str.contains(patron)].copy()
df_coincidencias['DESCRIPCION'].unique()

array(['Enseñanzas  Educación especial:  Educación Infantil Educación Básica Obligatoria Transición a la Vida Adulta  Ubicado en el Hospital Infantil de San Rafael',
       'ENSEÑANZA: Educación Infantil (Primer y segundo ciclo) - Primaria - ESO',
       'Enseñanza: Educación infantil (Primer y segundo ciclo)',
       'ENSEÑANZA: Educación infantil (Primer ciclo)',
       'Enseñanza: Educación infantil (Primer ciclo)',
       'ENSEÑANZA: Educación infantil (Primer Ciclo)',
       'Enseñanza: Educación infantil (Primer Ciclo)',
       'ENSEÑANZA: Sistema Educativo Alemán  EDUCACIÓN: Infantil 1er y 2&amp;ordm; Ciclo ALUMNOS: Españoles - Alemanes  IDIOMA: Alemán  EDAD: De 2 a 6 años',
       'Enseñanza:   Educación infantil (Primer ciclo)',
       'ENSEÑANZA: Educación infantil (Primer y segundo ciclo)',
       'Enseñanza: Educación infantil primer ciclo',
       'ENSEÑANZA: Educación infantil (primer y segundo ciclo)',
       'ENSEÑANZA: Educación infantil',
       'Enseñanza: Educación 

In [85]:
df_coincidencias[df_coincidencias['DESCRIPCION'].isna()]

NOMBRE DESCRIPCION  \
903               Escuela Infantil Contumami la Vaguada         NaN   
947   Escuela Infantil El Bosque Encantado. Calle An...         NaN   
988                        Escuela Infantil Fantasía II         NaN   
1057  Escuela Infantil Lapizitos. Calle Condesa de Teba         NaN   
1102                     Escuela Infantil Mamá Queca II         NaN   
1225                             Escuela Infantil Nubbe         NaN   
1227     Escuela Infantil Nubes. Calle Gabriel Usera 39         NaN   
1250                          Escuela Infantil Pequemun         NaN   
1256                   Escuela Infantil Pequeños Amigos         NaN   
1317   Escuela Infantil Smiles Bilingual Nursery School         NaN   
1338                          Escuela Infantil Trapitos         NaN   

                     NOMBRE-VIA CLASE-VIAL TIPO-NUM  NUM  CODIGO-POSTAL  \
903                  FONSAGRADA      PLAZA      NUM   10          28029   
947               ANTONIO LOPEZ      CALLE        V  246          28026   
988                JUSTA GARCIA      CALLE        V   12          28026   
1057            CONDESA DE TEBA      CALLE        V   11          28044   
1102           SANCHEZ GUERRERO      CALLE        V    7          28043   
1225                TRES OLIVOS      PLAZA      NUM   16          28034   
1227              GABRIEL USERA      CALLE        V   39          28026   
1250                     OLIVOS      CALLE        V    8          28003   
1256  MONASTERIO DE EL ESCORIAL    AVENIDA      NUM   74          28049   
1317            PUENTE LA REINA      CALLE        V   36          28050   
1338                   LIMONITA      CALLE        V    1          28021   

            BARRIO COD-DISTRITO             DISTRITO  COORDENADA-X  \
903       EL PILAR          8.0  FUENCARRAL-EL PARDO      439960.0   
947    ALMENDRALES         12.0                USERA      440926.0   
988     PRADOLONGO         12.0                USERA      439780.0   
1057    BUENAVISTA         11.0          CARABANCHEL      435616.0   
1102      CANILLAS         16.0            HORTALEZA      444636.0   
1225      VALVERDE          8.0  FUENCARRAL-EL PARDO      441536.0   
1227    PRADOLONGO         12.0                USERA      439858.0   
1250  VALLEHERMOSO          7.0             CHAMBERI      439198.0   
1256     EL GOLOSO          8.0  FUENCARRAL-EL PARDO      441218.0   
1317      VALVERDE          8.0  FUENCARRAL-EL PARDO      442593.0   
1338      BUTARQUE         17.0           VILLAVERDE      442684.0   

      COORDENADA-Y    LATITUD  LONGITUD  \
903      4481061.0  40.478058 -3.708338   
947      4470378.0  40.381906 -3.695938   
988      4470412.0  40.382114 -3.709464   
1057     4469483.0  40.373434 -3.758423   
1102     4479749.0  40.466573 -3.653061   
1225     4483695.0  40.501908 -3.689989   
1227     4470682.0  40.384561 -3.708571   
1250     4477334.0  40.444437 -3.716982   
1256     4484364.0  40.507912 -3.693804   
1317     4483689.0  40.501918 -3.677526   
1338     4466087.0  40.343366 -3.674854   

                                                   TIPO  
903   /contenido/entidadesYorganismos/CentrosEducaci...  
947   /contenido/entidadesYorganismos/CentrosEducaci...  
988   /contenido/entidadesYorganismos/CentrosEducaci...  
1057  /contenido/entidadesYorganismos/CentrosEducaci...  
1102  /contenido/entidadesYorganismos/CentrosEducaci...  
1225  /contenido/entidadesYorganismos/CentrosEducaci...  
1227  /contenido/entidadesYorganismos/CentrosEducaci...  
1250  /contenido/entidadesYorganismos/CentrosEducaci...  
1256  /contenido/entidadesYorganismos/CentrosEducaci...  
1317  /contenido/entidadesYorganismos/CentrosEducaci...  
1338  /contenido/entidadesYorganismos/CentrosEducaci...

In [86]:
df_coincidencias.fillna('ENSEÑANZA: Educacion infantil', inplace=True)

/tmp/ipykernel_62465/295666320.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value 'ENSEÑANZA: Educacion infantil' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df_coincidencias.fillna('ENSEÑANZA: Educacion infantil', inplace=True)


In [87]:
df_schools.update(df_coincidencias)

In [88]:
df_schools[df_schools['DESCRIPCION'].isna()]

NOMBRE DESCRIPCION  \
89                           Centro de enseñanza CODESA         NaN   
90                        Centro de enseñanza Cruz Roja         NaN   
91    Centro de enseñanza Cruz Roja Escuela Profesional         NaN   
146   Centro integrado de enseñanzas regladas a dist...         NaN   
1369                       ESERP Madrid Business School         NaN   

           NOMBRE-VIA CLASE-VIAL TIPO-NUM  NUM  CODIGO-POSTAL  \
89      DIEGO DE LEON      CALLE        V   34          28006   
90    DE RAFAEL VILLA      CALLE        V  NaN          28023   
91     REINA VICTORIA    AVENIDA        V   28          28003   
146             PRADO      PASEO        V   28          28014   
1369       COSTA RICA      CALLE        V    9          28016   

                    BARRIO COD-DISTRITO         DISTRITO  COORDENADA-X  \
89              CASTELLANA          4.0        SALAMANCA      442210.0   
90              EL PLANTIO          9.0  MONCLOA-ARAVACA      430541.0   
91    CIUDAD UNIVERSITARIA          9.0  MONCLOA-ARAVACA      439817.0   
146                 CORTES          1.0           CENTRO      441174.0   
1369          NUEVA ESPAÑA          5.0        CHAMARTIN      442773.0   

      COORDENADA-Y    LATITUD  LONGITUD  \
89       4476230.0  40.434707 -3.681346   
90       4480174.0  40.469333 -3.819366   
91       4477640.0  40.447246 -3.709695   
146      4473706.0  40.411896 -3.693334   
1369     4478893.0  40.458737 -3.674947   

                                                   TIPO  
89    /contenido/entidadesYorganismos/CentrosEducaci...  
90    /contenido/entidadesYorganismos/CentrosEducaci...  
91    /contenido/entidadesYorganismos/CentrosEducaci...  
146   /contenido/entidadesYorganismos/CentrosEducaci...  
1369  /contenido/entidadesYorganismos/CentrosEducaci...

El ESERP es una universidad, el codesa es una escuela de formación profesional, el Centro integrado de enseñanzas regladas a distancia parece algo gubernamental que realmente se escapa de las categorías comunes. Los centros de enseñanza de la cruz roja son ambos de FP


In [89]:
df_schools.loc[df_schools['NOMBRE'] == 'Centro de enseñanza CODESA', 'DESCRIPCION'] = 'Formación Profesional'
df_schools.loc[df_schools['NOMBRE'] == 'Centro de enseñanza Cruz Roja', 'DESCRIPCION'] = 'Formación Profesional'
df_schools.loc[df_schools[
                   'NOMBRE'] == 'Centro de enseñanza Cruz Roja Escuela Profesional', 'DESCRIPCION'] = 'Formación Profesional'
df_schools.loc[df_schools[
                   'NOMBRE'] == 'Centro integrado de enseñanzas regladas a distancia', 'DESCRIPCION'] = 'Centro integrado de enseñanzas regladas a distancia'
df_schools.loc[df_schools['NOMBRE'] == 'ESERP Madrid Business School', 'DESCRIPCION'] = 'Universidad ADE'

In [90]:
df_schools[df_schools['COORDENADA-X'].isna()]

NOMBRE  \
104  Centro de enseñanza Instede Caja Mágica   

                                           DESCRIPCION NOMBRE-VIA CLASE-VIAL  \
104  Formación profesional grado medio   Actividade...    PERALES     CAMINO   

    TIPO-NUM  NUM  CODIGO-POSTAL   BARRIO COD-DISTRITO DISTRITO  COORDENADA-X  \
104        V  NaN          28041  ORCASUR         12.0    USERA           NaN   

     COORDENADA-Y    LATITUD  LONGITUD  \
104           NaN  40.369242 -3.685088   

                                                  TIPO  
104  /contenido/entidadesYorganismos/CentrosEducaci...

In [91]:
df_schools[df_schools['COORDENADA-Y'].isna()]

NOMBRE  \
104  Centro de enseñanza Instede Caja Mágica   

                                           DESCRIPCION NOMBRE-VIA CLASE-VIAL  \
104  Formación profesional grado medio   Actividade...    PERALES     CAMINO   

    TIPO-NUM  NUM  CODIGO-POSTAL   BARRIO COD-DISTRITO DISTRITO  COORDENADA-X  \
104        V  NaN          28041  ORCASUR         12.0    USERA           NaN   

     COORDENADA-Y    LATITUD  LONGITUD  \
104           NaN  40.369242 -3.685088   

                                                  TIPO  
104  /contenido/entidadesYorganismos/CentrosEducaci...

No hay mucho problema, de todas formas ibamos a tirar los utm y quedarnos con lat long

In [92]:
df_schools[df_schools['LONGITUD'].isna()]

NOMBRE  \
463  Colegio Público de Educación Especial La Quinta   
783                                     Escuela Ideo   

                                           DESCRIPCION  \
463  Enseñanzas  Educación especial:  Educación inf...   
783  Enseñanzas   Primer Ciclo de Educación Infanti...   

                                NOMBRE-VIA CLASE-VIAL TIPO-NUM NUM  \
463  SOMONTES AL PALACIO DE LA REAL QUINTA  CARRETERA        V   1   
783                         COLMENAR VIEJO  CARRETERA        V  16   

     CODIGO-POSTAL    BARRIO COD-DISTRITO             DISTRITO  COORDENADA-X  \
463          28035  EL PARDO          8.0  FUENCARRAL-EL PARDO      436176.0   
783          28048       NaN          NaN  FUENCARRAL-EL PARDO      439661.0   

     COORDENADA-Y LATITUD LONGITUD  \
463     4482668.0     NaN      NaN   
783     4490699.0     NaN      NaN   

                                                  TIPO  
463  /contenido/entidadesYorganismos/CentrosEducaci...  
783  /contenido/entidadesYorganismos/CentrosEducaci...

In [93]:
df_schools[df_schools['LATITUD'].isna()]

NOMBRE  \
463  Colegio Público de Educación Especial La Quinta   
783                                     Escuela Ideo   

                                           DESCRIPCION  \
463  Enseñanzas  Educación especial:  Educación inf...   
783  Enseñanzas   Primer Ciclo de Educación Infanti...   

                                NOMBRE-VIA CLASE-VIAL TIPO-NUM NUM  \
463  SOMONTES AL PALACIO DE LA REAL QUINTA  CARRETERA        V   1   
783                         COLMENAR VIEJO  CARRETERA        V  16   

     CODIGO-POSTAL    BARRIO COD-DISTRITO             DISTRITO  COORDENADA-X  \
463          28035  EL PARDO          8.0  FUENCARRAL-EL PARDO      436176.0   
783          28048       NaN          NaN  FUENCARRAL-EL PARDO      439661.0   

     COORDENADA-Y LATITUD LONGITUD  \
463     4482668.0     NaN      NaN   
783     4490699.0     NaN      NaN   

                                                  TIPO  
463  /contenido/entidadesYorganismos/CentrosEducaci...  
783  /contenido/entidadesYorganismos/CentrosEducaci...

Existen dos entradas que no tienen su contrapartida en Lat long. Los solucionaré a mano.

In [94]:
lat, long = my_map_creator.utm_to_latlon(utm_x=439661.0, utm_y=4490699.0)
df_schools.loc[df_schools['NOMBRE'] == 'Escuela Ideo', 'LATITUD'] = lat
df_schools.loc[df_schools['NOMBRE'] == 'Escuela Ideo', 'LONGITUD'] = long

lat, long = my_map_creator.utm_to_latlon(utm_x=436176.0, utm_y=4482668.0)
df_schools.loc[df_schools['NOMBRE'] == 'Colegio Público de Educación Especial La Quinta', 'LATITUD'] = lat
df_schools.loc[df_schools['NOMBRE'] == 'Colegio Público de Educación Especial La Quinta', 'LONGITUD'] = long

In [95]:
df_schools.drop(columns=['COORDENADA-X', 'COORDENADA-Y'], inplace=True)

In [96]:
df_schools.head()

NOMBRE  \
0                             Academia Darío Estudio   
1                      Altair, Colegio Internacional   
2  Aula Municipal de Formación de Personas Adulta...   
3  Aula Municipal de Formación de Personas Adulta...   
4  Aula Municipal de Formación de Personas Adulta...   

                                         DESCRIPCION       NOMBRE-VIA  \
0  ENSEÑANZA: BACHILLERATO (Humanidades  y C. Soc...         GRAN VIA   
1  ENSEÑANZA: Educación infantil - Primaria - ESO...      JOAQUIN BAU   
2  Programas Formativos:   Español para extranjeros.           CASINO   
3  Programas Formativos:   LectoescrituraMatemáti...          ALMAGRO   
4  Programas Formativos:    Lectoescritura. Matem...  SANTA FELICIDAD   

  CLASE-VIAL TIPO-NUM NUM  CODIGO-POSTAL        BARRIO COD-DISTRITO  \
0      CALLE        V  63          28013       PALACIO          1.0   
1      CALLE      NUM   4          28036  NUEVA ESPAÑA          5.0   
2      CALLE        V   5          28005   EMBAJADORES          1.0   
3      CALLE        V   5          28010       ALMAGRO          7.0   
4      CALLE      NUM  39          28017        VENTAS         15.0   

        DISTRITO    LATITUD  LONGITUD  \
0         CENTRO  40.422491 -3.709818   
1      CHAMARTIN  40.463788 -3.685848   
2         CENTRO  40.406617 -3.705326   
3       CHAMBERI  40.428526 -3.695039   
4  CIUDAD LINEAL  40.424435 -3.650284   

                                                TIPO  
0  /contenido/entidadesYorganismos/CentrosEducaci...  
1  /contenido/entidadesYorganismos/CentrosEducaci...  
2  /contenido/entidadesYorganismos/CentrosEducaci...  
3  /contenido/entidadesYorganismos/CentrosEducaci...  
4  /contenido/entidadesYorganismos/CentrosEducaci...

Solo queda transformar la última columna y habremos terminado de limpiar

In [97]:
df_schools['TIPO'] = df_schools['TIPO'].str.extract(r'/([^/]+)$')

In [98]:
df_schools['TIPO'].head()

0    ColegiosPrivadosConcertados
1    ColegiosPrivadosConcertados
2        CentrosEducacionAdultos
3        CentrosEducacionAdultos
4        CentrosEducacionAdultos
Name: TIPO, dtype: object

# Análisis
Vamos a combinar los dos datasets para ver si existe alguna concentración de accidentes en alguna escuela y también observar qué tipo de accidente suele ocurrir de forma más frecuente.

In [99]:
df_accidentes.head()

num_expediente       fecha     hora  \
0    2022S000001  01/01/2022  1:30:00   
1    2022S000001  01/01/2022  1:30:00   
2    2022S000002  01/01/2022  0:30:00   
3    2022S000002  01/01/2022  0:30:00   
4    2022S000002  01/01/2022  0:30:00   

                                 localizacion numero  cod_distrito  \
0                          AVDA. ALBUFERA, 19     19          13.0   
1                          AVDA. ALBUFERA, 19     19          13.0   
2  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO      2           3.0   
3  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO      2           3.0   
4  PLAZA. CANOVAS DEL CASTILLO / PASEO. PRADO      2           3.0   

             distrito           tipo_accidente estado_meteorológico  \
0  PUENTE DE VALLECAS                  Alcance            Despejado   
1  PUENTE DE VALLECAS                  Alcance            Despejado   
2              RETIRO  Colisión fronto-lateral                  NaN   
3              RETIRO  Colisión fronto-lateral                  NaN   
4              RETIRO  Colisión fronto-lateral                  NaN   

             tipo_vehiculo  ...       rango_edad sexo cod_lesividad  \
0                  Turismo  ...  De 30 a 34 años    M           NaN   
1                  Turismo  ...  De 45 a 49 años    H           NaN   
2  Motocicleta hasta 125cc  ...  De 30 a 34 años    H           NaN   
3  Motocicleta hasta 125cc  ...  De 35 a 39 años    M           NaN   
4                  Turismo  ...  De 40 a 44 años    H           NaN   

   lesividad coordenada_x_utm coordenada_y_utm positiva_alcohol  \
0        NaN       443359,226      4472082,272              0.0   
1        NaN       443359,226      4472082,272              0.0   
2        NaN       441155,351      4474129,588              1.0   
3        NaN       441155,351      4474129,588              0.0   
4        NaN       441155,351      4474129,588              0.0   

   positiva_droga  numero_pasajeros  fugado  
0             0.0                 2      No  
1             0.0                 2      No  
2             0.0                 3      No  
3             0.0                 3      No  
4             0.0                 3      No  

[5 rows x 21 columns]

Vamos a necesitar una columna nueva en df_schools que refleje una localización completa

In [100]:
# def tiene_coincidencia_palabra(street_name, location):
#     words_nombre_via = street_name.split() 
#     words_location = location.split() 
#     # return any(palabra in words_location for palabra in words_nombre_via)

In [101]:
# df_accidentes['coincidencia'] = df_accidentes['localizacion'].apply(lambda x: df_schools['NOMBRE-VIA'].apply(lambda y: tiene_coincidencia_palabra(x, y)).any())

In [102]:
# df_accidentes[df_accidentes['coincidencia']]

Lo anterior no ha funcionado, ha sido demasiado generoso y habrá hecho match con preposiciones y determinantes. Vamos a probar a trabajar localización en accidentes antes de probar el match otra vez

In [103]:
def eliminar_primera_palabra(frase):
    palabras = frase.split()
    if len(palabras) > 1:
        return ' '.join(palabras[1:])
    else:
        return ""


df_accidentes['localizacion'] = df_accidentes['localizacion'].apply(eliminar_primera_palabra)

In [104]:
df_accidentes['localizacion'].head()

0                           ALBUFERA, 19
1                           ALBUFERA, 19
2    CANOVAS DEL CASTILLO / PASEO. PRADO
3    CANOVAS DEL CASTILLO / PASEO. PRADO
4    CANOVAS DEL CASTILLO / PASEO. PRADO
Name: localizacion, dtype: object

In [105]:
df_schools.rename(columns={'NOMBRE-VIA': 'localizacion'}, inplace=True)

In [106]:
df_coincidencia = df_accidentes.merge(df_schools, on='localizacion', how='inner')

In [107]:
df_coincidencia.drop_duplicates(subset=['num_expediente', 'tipo_vehiculo', 'tipo_persona', 'rango_edad', 'sexo', 'tipo_accidente', 'distrito', 'estado_meteorológico', 'numero_pasajeros'], inplace=True)

In [108]:
df_coincidencia

num_expediente       fecha      hora                  localizacion numero  \
0      2022S000807  15/01/2022  18:30:00                        ATOCHA    105   
2      2022S000807  15/01/2022  18:30:00                        ATOCHA    105   
4      2022S001047  18/01/2022  17:20:00  DOCTOR FEDERICO RUBIO Y GALI     60   
7      2022S001047  18/01/2022  17:20:00  DOCTOR FEDERICO RUBIO Y GALI     60   
10     2022S003067  10/02/2022   7:20:00        CANILLEJAS A VICALVARO     65   
..             ...         ...       ...                           ...    ...   
235    2023S027671  20/08/2023  16:48:00                      DELICIAS     34   
238    2023S027671  20/08/2023  16:48:00                      DELICIAS     34   
241    2023S027671  20/08/2023  16:48:00                      DELICIAS     34   
244    2023S027671  20/08/2023  16:48:00                      DELICIAS     34   
247    2023S027671  20/08/2023  16:48:00                      DELICIAS     34   

     cod_distrito             distrito           tipo_accidente  \
0             1.0               CENTRO         Colisión lateral   
2             1.0               CENTRO         Colisión lateral   
4             6.0               TETUÁN                    Caída   
7             6.0               TETUÁN                    Caída   
10           20.0  SAN BLAS-CANILLEJAS         Colisión lateral   
..            ...                  ...                      ...   
235           2.0           ARGANZUELA  Colisión fronto-lateral   
238           2.0           ARGANZUELA  Colisión fronto-lateral   
241           2.0           ARGANZUELA  Colisión fronto-lateral   
244           2.0           ARGANZUELA  Colisión fronto-lateral   
247           2.0           ARGANZUELA  Colisión fronto-lateral   

    estado_meteorológico            tipo_vehiculo  ... CLASE-VIAL TIPO-NUM  \
0                    NaN                  Turismo  ...      CALLE        V   
2                    NaN                  Turismo  ...      CALLE        V   
4              Despejado  Motocicleta hasta 125cc  ...    AVENIDA        V   
7              Despejado                  Turismo  ...    AVENIDA        V   
10                   NaN             Tractocamión  ...    AVENIDA      NUM   
..                   ...                      ...  ...        ...      ...   
235            Despejado                  Turismo  ...      PASEO      NUM   
238            Despejado                  Turismo  ...      PASEO      NUM   
241            Despejado                  Turismo  ...      PASEO      NUM   
244            Despejado                  Turismo  ...      PASEO      NUM   
247            Despejado                  Turismo  ...      PASEO      NUM   

     NUM  CODIGO-POSTAL         BARRIO COD-DISTRITO             DISTRITO  \
0     93          28012         CORTES          1.0               CENTRO   
2     93          28012         CORTES          1.0               CENTRO   
4      1          28039  BELLAS VISTAS          6.0               TETUAN   
7      1          28039  BELLAS VISTAS          6.0               TETUAN   
10   135          28022          ROSAS         20.0  SAN BLAS-CANILLEJAS   
..   ...            ...            ...          ...                  ...   
235   67          28045       DELICIAS          2.0           ARGANZUELA   
238   67          28045       DELICIAS          2.0           ARGANZUELA   
241   67          28045       DELICIAS          2.0           ARGANZUELA   
244   67          28045       DELICIAS          2.0           ARGANZUELA   
247   67          28045       DELICIAS          2.0           ARGANZUELA   

       LATITUD  LONGITUD                                            TIPO  
0    40.411081  -3.69634                         CentrosEducacionAdultos  
2    40.411081  -3.69634                         CentrosEducacionAdultos  
4    40.447766 -3.708453  CentrosFormacionProfesionalPrivadosConcertados  
7    40.447766 -3.708453  CentrosFormacionProfesionalPrivadosConcertados  
1

Vamos a ver en primer lugar si el tipo de accidente: Atropello a persona es más frecuente que en la media del dataset

In [109]:
import math

# Calcular la frecuencia de "Atropello a persona"
frecuencia_atropello_total = len(
    df_accidentes[df_accidentes['tipo_accidente'] == "Atropello a persona"]['tipo_accidente'])
frecuencia_atropello_colegios = len(
    df_coincidencia[df_coincidencia['tipo_accidente'] == "Atropello a persona"]['tipo_accidente'])

# Calcular el promedio de atropellos en tódo el dataset y en las coincidencias
promedio_atropello_total = frecuencia_atropello_total / len(df_accidentes)
promedio_atropello_colegios = frecuencia_atropello_colegios / len(df_coincidencia)

# Calcular la varianza para la proporción de atropellos como si fuera una variable binomial: O se atropella o no se atropella
varianza_atropello_total = (promedio_atropello_total * (1 - promedio_atropello_total)) / len(df_accidentes)
varianza_atropello_colegios = (promedio_atropello_colegios * (1 - promedio_atropello_colegios)) / len(df_coincidencia)

# Calcular la desviación estándar para el conjunto total de datos
desviacion_estandar_total = math.sqrt(varianza_atropello_total)
desviacion_estandar_colegios = math.sqrt(varianza_atropello_colegios)

# Mostrar los resultados
print("Frecuencia de atropellos total:", frecuencia_atropello_total, "  total de accidentes: ", len(df_accidentes))
print("Frecuencia de atropellos en colegios:", frecuencia_atropello_colegios, "  total de accidentes: ", len(df_coincidencia))
print("Promedio (proporción) de 'Atropello a persona' en total:", promedio_atropello_total)
print("Promedio (proporción) de 'Atropello a persona' en colegios:", promedio_atropello_colegios)
print("Desviaciones binomiales (se atropella o no se atropella)")
print("Desviación estándar para el conjunto total de datos:", desviacion_estandar_total)
print("Desviación estándar para colegios:", desviacion_estandar_colegios)


Frecuencia de atropellos total: 5264   total de accidentes:  74961
Frecuencia de atropellos en colegios: 6   total de accidentes:  105
Promedio (proporción) de 'Atropello a persona' en total: 0.07022318272168195
Promedio (proporción) de 'Atropello a persona' en colegios: 0.05714285714285714
Desviaciones binomiales (se atropella o no se atropella)
Desviación estándar para el conjunto total de datos: 0.000933279927821891
Desviación estándar para colegios: 0.022652141919202624


### Frecuencia de Atropellos Total:
Hubo 5,264 atropellos a personas dentro del total de 74,961 accidentes registrados. 

### Frecuencia de Atropellos en Colegios:
Dentro de 250 accidentes registrados en colegios, 6 fueron atropellos a personas. Aunque la muestra de accidentes en colegios es mucho menor en comparación con el total de accidentes, podemos observar que al menos durante el periodo estudiado, cerca de los colegios hubo menos atropellos que en la ciudad en general

### Promedio (Proporción) de 'Atropello a persona':
En el total de accidentes, aproximadamente el 7.02% (0.0702) fueron atropellos a personas. Esto te da una idea de cuán comunes son los atropellos a personas en relación con otros tipos de accidentes en el conjunto de datos completo.
En los colegios, aproximadamente el 5.7% (0.057) de los accidentes fueron atropellos a personas, una proporción que es menor que en el conjunto de datos completo.

### Desviación Estándar (Desviaciones Binomiales):
La desviación estándar para el conjunto total de datos es 0.000933, lo que indica una baja variabilidad o dispersión en la proporción de atropellos a personas respecto a la media en el conjunto de datos completo. Esto sugiere que la proporción de atropellos a personas es relativamente estable en el conjunto total de accidentes.
La desviación estándar para los colegios es 0.022, que es mayor que la del conjunto total de datos. Esto indica una mayor variabilidad o dispersión en la proporción de atropellos a personas en colegios en comparación con el conjunto total de datos. Aunque la diferencia en proporciones no es muy grande, la mayor desviación estándar sugiere que la proporción de atropellos en colegios puede ser más inconsistente o variar más de un conjunto de datos a otro en comparación con la proporción en el conjunto total de accidentes.

En resumen, aunque la proporción de atropellos a personas es ligeramente más alta en colegios que en el conjunto total de accidentes, la variabilidad de esta proporción también es mayor en colegios, lo que indica que la frecuencia de atropellos en colegios puede ser más sujeta a fluctuaciones o variaciones más amplias.


In [110]:
import statsmodels.api as sm

# Número de atropellos y tamaño de la muestra total
num_atropellos_total = frecuencia_atropello_total
tamano_muestra_total = len(df_accidentes)

# Proporción de atropellos en el conjunto total de datos
proporcion_atropellos_total = num_atropellos_total / tamano_muestra_total

# Calcular el intervalo de confianza
conf_int_total = sm.stats.proportion_confint(num_atropellos_total, tamano_muestra_total, alpha=0.05, method='wilson')

print("Proporción de 'Atropello a persona' en el conjunto total de datos:", proporcion_atropellos_total)
print("Intervalo de confianza del 95%:", conf_int_total)

# Número de atropellos y tamaño de la muestra en colegios
num_atropellos_colegios = frecuencia_atropello_colegios
tamaño_muestra_colegios = len(df_coincidencia)

# Proporción de atropellos en el conjunto de datos de colegios
proporcion_atropellos_colegios = num_atropellos_colegios / tamaño_muestra_colegios

# Calcular el intervalo de confianza
conf_int_colegios = sm.stats.proportion_confint(num_atropellos_colegios, tamaño_muestra_colegios, alpha=0.05, method='wilson')

print("Proporción de 'Atropello a persona' en el conjunto de datos de colegios:", proporcion_atropellos_colegios)
print("Intervalo de confianza del 95% para colegios:", conf_int_colegios)

Proporción de 'Atropello a persona' en el conjunto total de datos: 0.07022318272168195
Intervalo de confianza del 95%: (0.0684159252230113, 0.07207448673371901)
Proporción de 'Atropello a persona' en el conjunto de datos de colegios: 0.05714285714285714
Intervalo de confianza del 95% para colegios: (0.02644963443732167, 0.11909654798560423)


La estimación puntual de la proporción en colegios es del 5.7%, mientras que en el conjunto total de datos es del 7.02%. Además, el intervalo de confianza del 95% para colegios (2.64% - 11.90%) es más amplio y abarca un rango más grande de valores en comparación con el intervalo de confianza del 95% para el conjunto total de datos (6.84% - 7.21%). Esto sugiere que la proporción de atropellos a personas en colegios podría ser significativamente mayor (o menor!) en comparación con el conjunto total de accidentes, pero debido a la variabilidad en la muestra de colegios, no podemos estar tan seguros de la magnitud exacta de la diferencia.

Veamos qué pasa con los tipos de lesiones en cerca de los colegios

In [111]:
df_lesividad_schools = df_coincidencia.groupby('lesividad').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_lesividad_schools.rename(columns={'num_expediente': 'proportion_schools'}, inplace=True)

In [112]:
df_lesividad_total = df_accidentes.groupby('lesividad').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_lesividad_total.rename(columns={'num_expediente': 'proportion_totals'}, inplace=True)

In [113]:
df_lesividad_schools['proportion_schools'] = df_lesividad_schools.proportion_schools / sum(df_lesividad_schools['proportion_schools'])
df_lesividad_total['proportion_totals'] = df_lesividad_total.proportion_totals / sum(df_lesividad_total['proportion_totals'])

In [114]:
df_lesividad_schools

lesividad  proportion_schools
0  Asistencia sanitaria inmediata en centro de sa...            0.014925
1        Atención en urgencias sin posterior ingreso            0.014925
2  Asistencia sanitaria ambulatoria con posterior...            0.059701
3                Ingreso inferior o igual a 24 horas            0.089552
4  Asistencia sanitaria sólo en el lugar del acci...            0.164179
5                           Sin asistencia sanitaria            0.656716

In [115]:
df_lesividad_total

lesividad  proportion_totals
0                                       Se desconoce           0.000073
1                                 Fallecido 24 horas           0.000980
2                        Ingreso superior a 24 horas           0.021182
3  Asistencia sanitaria ambulatoria con posterior...           0.022284
4  Asistencia sanitaria inmediata en centro de sa...           0.041923
5        Atención en urgencias sin posterior ingreso           0.056248
6                Ingreso inferior o igual a 24 horas           0.074099
7  Asistencia sanitaria sólo en el lugar del acci...           0.240297
8                           Sin asistencia sanitaria           0.542915

En el primer gráfico, que representa los accidentes cerca de escuelas, las categorías con mayor frecuencia son:

- "Sin asistencia sanitaria" (aproximadamente 65%)
- "Asistencia sanitaria sólo en el lugar del accidente" (aproximadamente 16%)
- "Ingreso inferior o igual a 24 horas" (aproximadamente 9%)
- "Asistencia sanitaria ambulatoria con posterioridad" (aproximadamente 6%)


Esto sugiere que la mayoría de los accidentes cerca de las escuelas no resultan en asistencia sanitaria adicional o solo requieren atención en el lugar del accidente. Una posible interpretación es que los accidentes cerca de las escuelas tienden a ser menos graves o que hay una tendencia a no buscar asistencia sanitaria posterior.

En el segundo gráfico, que representa la totalidad de los accidentes en Madrid, las categorías más frecuentes son:

- "Sin asistencia sanitaria" (aproximadamente 54%)
- "Asistencia sanitaria sólo en el lugar del accidente" (aproximadamente 24%)
- "Atención en urgencias sin posterior ingreso" (aproximadamente 6%)
- "Asistencia sanitaria inmediata en centro de salud o mutua" (aproximadamente 4%)
- "Asistencia sanitaria ambulatoria con posterioridad" (aproximadamente 2.2%)

Se observa una tendencia similar en la distribución de las categorías, aunque hay una proporción bastante menor de casos sin asistencia sanitaria y una mayor proporción de casos que reciben asistencia sólo en el lugar del accidente en comparación con los accidentes cerca de las escuelas. Además, se ve un incremento en la categoría de "Atención en urgencias sin posterior ingreso".

Comparando ambos gráficos, se puede concluir que la proporción de asistencia sanitaria es similar en ambos contextos, aunque hay una tendencia bastante mayor a no requerir asistencia sanitaria en los accidentes cerca de escuelas. Esto podría sugerir que los accidentes cerca de las escuelas son menos graves en comparación con el promedio de la ciudad o que hay un mayor seguimiento de los accidentes en otras áreas de Madrid que resulta en más atenciones en urgencias. Donde se observa una clara diferencia es en la categoría "Asistencia sanitaria ambulatoria con posterioridad" aproximadamente 5.9% cerca de escuelas y 2.2% en la ciudad en general esto puede deberse a pequeñas lesiones del accidente. 

Por lo tanto, es menos probable necesitar atención sanitaria en una accidente cerca de una escuela y cuando se necesita atención sanitaria, suele ser por lesiones de menor gravedad que en el promedio de la ciudad

Creemos que puede deberse a las zonas con menor velocidad y a la debida señalización de las zonas escolares.


Vamos a estudiar la tendencia horaria de los accidentes

In [123]:
df_time_schools = df_coincidencia.groupby('hora').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_time_schools.rename(columns={'num_expediente': 'proportion_schools'}, inplace=True)

In [124]:
df_time_schools['proportion_schools'] = df_time_schools.proportion_schools / sum(df_time_schools['proportion_schools'])

In [125]:
df_time_schools

hora  proportion_schools
0    0:15:00            0.009524
1    1:02:00            0.009524
2   17:15:00            0.009524
3    1:30:00            0.009524
4   21:45:00            0.009524
5   16:20:00            0.009524
6    3:20:00            0.009524
7   15:25:00            0.009524
8   11:58:00            0.009524
9    0:32:00            0.009524
10   5:30:00            0.009524
11  23:20:00            0.019048
12   9:05:00            0.019048
13  23:10:00            0.019048
14   7:20:00            0.019048
15  22:32:00            0.019048
16   8:22:00            0.019048
17  20:24:00            0.019048
18   8:30:00            0.019048
19   8:55:00            0.019048
20  18:44:00            0.019048
21   9:30:00            0.019048
22  15:19:00            0.019048
23  17:20:00            0.019048
24  17:00:00            0.019048
25  10:00:00            0.019048
26  10:35:00            0.019048
27  10:55:00            0.019048
28  12:05:00            0.019048
29  16:30:00            0.019048
30  16:00:00            0.019048
31  18:30:00            0.019048
32  14:30:00            0.019048
33   0:34:00            0.028571
34   8:40:00            0.028571
35  19:00:00            0.028571
36  20:15:00            0.028571
37  19:03:00            0.028571
38  23:55:00            0.028571
39  16:38:00            0.038095
40  17:25:00            0.038095
41  23:45:00            0.057143
42  16:48:00            0.057143
43  20:04:00            0.057143
44   1:23:00            0.057143

In [126]:
df_time_schools = df_accidentes.groupby('hora').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_time_schools.rename(columns={'num_expediente': 'proportion_schools'}, inplace=True)
df_time_schools['proportion_schools'] = df_time_schools.proportion_schools / sum(df_time_schools['proportion_schools'])

In [127]:
df_time_schools

hora  proportion_schools
0      4:36:00            0.000013
1      5:38:00            0.000013
2      7:01:00            0.000013
3      1:33:00            0.000013
4     11:16:00            0.000013
...        ...                 ...
1325  20:00:00            0.008004
1326  16:00:00            0.008098
1327  19:00:00            0.008178
1328  15:30:00            0.008538
1329  18:00:00            0.008871

[1330 rows x 2 columns]

No se ve una clara correlación. Exploramos el dato raro 23:45:00, 0.120 de los accidentes de los colegios un dato muy inusual

In [121]:
df_coincidencia[df_coincidencia['hora'] == "23:45:00"]

num_expediente       fecha      hora            localizacion numero  \
20    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   
25    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   
30    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   
35    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   
40    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   
45    2023S021227  28/05/2023  23:45:00  CANILLEJAS A VICALVARO    47B   

    cod_distrito             distrito           tipo_accidente  \
20          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   
25          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   
30          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   
35          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   
40          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   
45          20.0  SAN BLAS-CANILLEJAS  Colisión fronto-lateral   

   estado_meteorológico tipo_vehiculo  ... CLASE-VIAL TIPO-NUM  NUM  \
20            Despejado     Furgoneta  ...    AVENIDA      NUM  135   
25            Despejado     Furgoneta  ...    AVENIDA      NUM  135   
30            Despejado       Turismo  ...    AVENIDA      NUM  135   
35            Despejado       Turismo  ...    AVENIDA      NUM  135   
40            Despejado       Turismo  ...    AVENIDA      NUM  135   
45            Despejado       Turismo  ...    AVENIDA      NUM  135   

    CODIGO-POSTAL BARRIO COD-DISTRITO             DISTRITO    LATITUD  \
20          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   
25          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   
30          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   
35          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   
40          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   
45          28022  ROSAS         20.0  SAN BLAS-CANILLEJAS  40.426565   

    LONGITUD                         TIPO  
20 -3.610022  ColegiosPrivadosConcertados  
25 -3.610022  ColegiosPrivadosConcertados  
30 -3.610022  ColegiosPrivadosConcertados  
35 -3.610022  ColegiosPrivadosConcertados  
40 -3.610022  ColegiosPrivadosConcertados  
45 -3.610022  ColegiosPrivadosConcertados  

[6 rows x 33 columns]

creo que es una entrada repetida 30 veces por alguna razón

aqui me di cuenta de mi error de no hacer inplace en el drop_duplicates

Vamos a ver si el tiempo afecta en algo

In [132]:
df_weather_schools = df_coincidencia.groupby('estado_meteorológico').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_weather_schools.rename(columns={'num_expediente': 'proportion_schools'}, inplace=True)
df_weather_schools['proportion_schools'] = df_weather_schools.proportion_schools / sum(df_weather_schools['proportion_schools'])

In [133]:
df_weather_schools

estado_meteorológico  proportion_schools
0       LLuvia intensa            0.021978
1         Se desconoce            0.032967
2              Nublado            0.054945
3         Lluvia débil            0.120879
4            Despejado            0.769231

In [134]:
df_weather_totals = df_accidentes.groupby('estado_meteorológico').count().sort_values(by='num_expediente')['num_expediente'].reset_index()
df_weather_totals.rename(columns={'num_expediente': 'proportion_schools'}, inplace=True)
df_weather_totals['proportion_schools'] = df_weather_totals.proportion_schools / sum(df_weather_totals['proportion_schools'])

In [135]:
df_weather_totals

estado_meteorológico  proportion_schools
0              Nevando            0.000090
1           Granizando            0.000195
2         Se desconoce            0.006877
3       LLuvia intensa            0.014339
4         Lluvia débil            0.059693
5              Nublado            0.069716
6            Despejado            0.849091

Según nuestro análisis anterior, sabemos que en este periodo hubo un promedio de:


In [ ]:
# todo continuar aquí.
# me he ido a hacer lo del tiempo